x-pack/plugins/case/server/routes/api/update_case.ts

```diff
@@ -17,23 +24,61 @@ export function initUpdateCaseApi({ caseService, router }: RouteDeps) {
         params: schema.object({
           id: schema.string(),
         }),
-        body: UpdatedCaseSchema,
+        body: UpdateCaseArguments,
       },
     },
     async (context, request, response) => {
+      let theCase: SavedObject<CaseAttributes>;
       try {
-        const updatedCase = await caseService.updateCase({
+        theCase = await caseService.getCase({
           client: context.core.savedObjects.client,
           caseId: request.params.id,
-          updatedAttributes: {
-            ...request.body,
-            updated_at: new Date().toISOString(),
-          },
         });
-        return response.ok({ body: updatedCase.attributes });
       } catch (error) {
         return response.customError(wrapError(error));
       }
+
+      if (request.body.version !== theCase.version) {
+        return response.customError(
+          wrapError(
+            Boom.conflict(
+              'This case has been updated. Please refresh before saving additional updates.'
+            )
+          )
+        );
+      }
+      const updateCase: UpdateCase = request.body.case;
+      const currentCase = theCase.attributes;
+      Object.keys(updateCase).forEach(key => {
+        if (
+          key === 'tags' &&
+          updateCase.tags &&
```

Comment:
nit: Nothing really important, I assume for in the future we might have other fields the take an array, so I think it would make more sense if we put updateCase[key]

Reply:
what about doing something like that? and using `difference` from lodash/fp
```
const updateCase: Partial<UpdateCase> = Object.entries(request.body.case).reduce(
        (acc, [key, value]) => {
          const currentValue = currentCase[key];
          if (
            Array.isArray(value) &&
            Array.isArray(currentValue) &&
            difference(value, currentValue).length !== 0
          ) {
            return {
              ...acc,
              [key]: value,
            };
          } else if (value !== currentCase[key]) {
            return {
              ...acc,
              [key]: value,
            };
          }
          return acc;
        },
        {}
      );
```